geoopt/manifolds/lorentz/__init__.py

```diff
@@ -0,0 +1,221 @@
+import torch as th
+import torch.nn
+import numpy as np
+from typing import Tuple, Optional
+from . import math
+import geoopt
+from ..base import Manifold, ScalingInfo
+from ...utils import size2shape, broadcast_shapes
+
+__all__ = ["Lorentz"]
+
+_lorentz_ball_doc = r"""
+    Hyperboloid model
+
+    Parameters
+    ----------
+    k : float|tensor
+        manifold negative curvature
+
+    Notes
+    -----
+    It is extremely recommended to work with this manifold in double precision
+"""
+
+
+class Lorentz(Manifold):
+    __doc__ = r"""{}
+    """
+
+    ndim = 1
+    reversible = False
+    name = "Hyperboloid"
+    __scaling__ = Manifold.__scaling__.copy()
+
+    def __init__(self, k=1.0):
+        super().__init__()
+        self.register_buffer("k", torch.as_tensor(k, dtype=torch.get_default_dtype()))
```

Comment:
Btw, you can make it learnable as done in Poincare.

Reply:
You sure poincare has learnable params?